pkg/threescale/threescale.go

```diff
@@ -140,9 +146,16 @@ func (s *Threescale) isAuthorized(svcID string, request authorization.InstanceMs
 		appID = request.Subject.Properties["app_id"].GetStringValue()
 		appKey = request.Subject.Properties["app_key"].GetStringValue()
 
+		clientID = request.Subject.Properties["client_id"].GetStringValue()
+
 		userKey = request.Subject.User
 	}
 
+	if proxyConf.Content.BackendVersion == openIDTypeIdentifier {
+		// OIDC integration configured so force app identifier to come from jwt claims
+		appID = clientID
```

Comment:
this would only make sense in the above `if` clause, as you can't have a `clientID` without a valid `request.Subject` - which also means you can essentially remove `clientID` and have this assigned as below - wdyt?
```golang
  if request.Subject != nil {
    var appIDKey

    if proxyConf.Content.BackendVersion == openIDTypeIdentifier {
      appIDKey = "client_id"
    }  else {
      appIDKey = "app_id"
      // other code ...
      // userKey = ...
    }

    appID = request.Subject.Properties[app_id_key].GetStringValue()
    appKey = ...
  }
```